slobrok/src/vespa/slobrok/server/remote_slobrok.cpp

```diff
@@ -33,8 +33,7 @@ RemoteSlobrok::RemoteSlobrok(const std::string &name, const std::string &spec,
     _rpcserver.healthCheck();
 }
 
-RemoteSlobrok::~RemoteSlobrok()
-{
+void RemoteSlobrok::shutdown() {
```

Comment:
(not really a comment, just some random thoughts)

If shutdown is called from any other thread than the network thread; the RPC cleanup does not seem safe. If we remove a peer as a result of re-config, that also need to be done in the network thread. Currently, we pretty much need to use a single thread for everything. We should consider just embracing it and not try to be thread-safe 'some places', since we already have identified possible self-deadlock scenarios with 2-way event listening that would probably be unproblematic if we just avoid locking.